android-app/app/src/androidTest/java/de/dhbw/ka/se/fibo/createAccount/CreateAccountTest.java

```diff
@@ -0,0 +1,166 @@
+package de.dhbw.ka.se.fibo.createAccount;
+
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static androidx.test.espresso.action.ViewActions.typeText;
+import static androidx.test.espresso.assertion.ViewAssertions.doesNotExist;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.hasErrorText;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.hamcrest.core.IsNot.not;
+
+import android.content.Context;
+import android.util.Log;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import de.dhbw.ka.se.fibo.CreateAccountActivity;
+import de.dhbw.ka.se.fibo.R;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
+
+
+public class CreateAccountTest {
+
+    /**
+     * Use {@link ActivityScenarioRule} to create and launch the activity under test, and close it
+     * after test completes.
+     */
+    @Rule
+    public ActivityScenarioRule<CreateAccountActivity> activityScenarioRule
+            = new ActivityScenarioRule<>(CreateAccountActivity.class);
+
+    private MockWebServer server = new MockWebServer();
+
+    @Before
+    public void setUp() throws IOException {
+        server.start(8000);
+    }
+
+    @After
+    public void tearDown() throws IOException {
+        server.close();
+    }
+
+    @Test
+    public void testCreateAccountButtonClick() {
+
+        onView(withId(R.id.create_account_button))
+                .perform(click());
+    }
+
+    @Test
+    public void testEmailInput() {
+
+        onView(withId(R.id.create_account_email))
+                .perform(typeText("test"), closeSoftKeyboard());
+
+        onView(withId(R.id.create_account_button))
+                .perform(click());
+    }
+
+    @Test
+    public void testPasswordWithoutInput() {
+
+        onView(withId(R.id.create_account_password))
+                .perform(typeText(""), closeSoftKeyboard());
+
+        onView(withId(R.id.create_account_button))
+                .perform(click());
+    }
+
+    @Test
+    public void testPasswordWithInput() {
+
+        onView(withId(R.id.create_account_password))
+                .perform(typeText("AsdfJklo1.2"), closeSoftKeyboard());
+
+        onView(withId(R.id.create_account_button))
+                .perform(click());
+    }
```

Comment:
Here you still need to add the tests it self. Currently you are not testing if the error messages are shown, instead espresso just types/clicks and the given points and finishes. Before finishing it should check if the necessary errormessage is displayed.

I'm also not sure what you want to test in the testCreateAccountButtonClick() method, but it looks like it was used to get started with espresso and can be deleted for now :)

Reply:
@Cebox82 can you handle this? :)